### PR TITLE
dev-python/statsmodels: version bump 0.10.1

### PR DESCRIPTION
**Pull request by TheChymera**
Package-Manager: Portage-2.3.75, Repoman-2.3.17
Signed-off-by: Horea Christian <chr@chymera.eu>

### CODE REVIEW

File: dev-python/statsmodels/statsmodels-0.10.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	distutils-r1_python_prepare_all
+}
+
+python_compile_all() {
+	use doc && esetup.py build_sphinx -b html --build-dir=docs/build
+}
+
+python_test() {
+	cd "${BUILD_DIR}" || die
+	python -c 'import statsmodels; statsmodels.test()' || die
```
Comment (juippis) [inline dev-python/statsmodels/statsmodels-0.10.1.ebuild]: Shouldn't you use `${EPYTHON}}` here?

File: dev-python/statsmodels/statsmodels-0.10.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python2_7 python3_{5,6,7} )
```
Comment (juippis) [inline dev-python/statsmodels/statsmodels-0.10.1.ebuild]: We usually have one empty line below EAPI.

File: dev-python/statsmodels/statsmodels-0.10.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+python_test() {
+	cd "${BUILD_DIR}" || die
+	python -c 'import statsmodels; statsmodels.test()' || die
+}
+
+python_install_all() {
+	find . -name \*LICENSE.txt -delete || die
+	use doc && HTML_DOCS=( docs/build/html/. )
```
Comment (juippis) [inline dev-python/statsmodels/statsmodels-0.10.1.ebuild]: It doesn't matter much, but `local HTML_DOCS` please (to follow PMS).

File: dev-python/statsmodels/statsmodels-0.10.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		dev-python/matplotlib[${PYTHON_USEDEP}]
+		dev-python/nbconvert[${PYTHON_USEDEP}]
+		dev-python/nbformat[${PYTHON_USEDEP}]
+		dev-python/numpydoc[${PYTHON_USEDEP}]
+		dev-python/sphinx[${PYTHON_USEDEP}]
+	)
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/nose[${PYTHON_USEDEP}]
+		)
```
Comment (juippis) [inline dev-python/statsmodels/statsmodels-0.10.1.ebuild]: Mis-indent.

**Comment by juippis** [comment]
I didnt have time to test this thorougly, but please check if sphinx modules are imported during build or if it just gets executed. the python2.7 dep for sphinx makes it impossible/annoying to solve. (Ie, remove PYTHON_USEDEP if it just gets executed)

**Comment by juippis** [comment]
Also, 
```
 * Messages for package dev-python/statsmodels-0.10.1:

 *   dev-python/matplotlib for Plotting functionality
 * The ebuild is installing to one or more unexpected paths:
 * 
 *   /usr/LICENSE.txt
 *   /usr/setup.cfg
```
```
>: file /usr/setup.cfg 
/usr/setup.cfg: ASCII text
```

**Comment by TheChymera** [comment]
> I didnt have time to test this thorougly, but please check if sphinx modules are imported during build or if it just gets executed. the python2.7 dep for sphinx makes it impossible/annoying to solve. (Ie, remove PYTHON_USEDEP if it just gets executed)

how can I check this?

**Comment by TheChymera** [comment]
As for those files, this is really strange, since they are also installed in the proper places.... should I just `rm` them at the beginning of `python_install_all()`?

```
chymera@darkhost ~ $ equery f statsmodels | ag LICENSE
/usr/LICENSE.txt
/usr/lib64/python2.7/site-packages/statsmodels/LICENSE.txt
/usr/lib64/python2.7/site-packages/statsmodels/stats/libqsturng/LICENSE.txt
/usr/lib64/python3.6/site-packages/statsmodels/LICENSE.txt
/usr/lib64/python3.6/site-packages/statsmodels/stats/libqsturng/LICENSE.txt
chymera@darkhost ~ $ equery f statsmodels | ag setup
/usr/lib64/python2.7/site-packages/statsmodels/setup.cfg
/usr/lib64/python3.6/site-packages/statsmodels/setup.cfg
/usr/setup.cfg
```

**Comment by juippis** [comment]
> 
> how can I check this?

https://github.com/statsmodels/statsmodels/blob/master/docs/Makefile
It's just executed. Basically you want to see if they use sphinx-build command to generate docs, or if the source files have `import sphinx` or similar.

As for the 2nd, Gentoo doesnt install LICENSEs there. Is the correct license included in portage's licenses/ folder? You can also remove them in src_install.

**Comment by TheChymera** [comment]
@juippis yes, ofc, it's the BSD license, should I remove all the license files entirely then?

What about `setup.cfg`?

**Comment by juippis** [comment]
I'm kinda 50-50 on the license thing. On the other hand, fast, I couldnt find a rule saying it has to be removed, but when grepping though my /usr there aren't _many_ LICENSE.txt files installed either. 

I'd say leave it for now, and leave setup.cfg if it provides documentation or functionality, but remove them from the top-level of /usr.

**Comment by juippis** [comment]
Are the FHS-related things fixed in your latest commit? It took me 45 minutes to compile this, I'd hate to wait that time only to notice the same errors :P

**Comment by TheChymera** [comment]
@juippis I'm close to giving up,

```
python_install_all() {
    pushd .. > /dev/null || die
        find . -wholename "*/statsmodels/LICENSE.txt" -delete || die
        find . -wholename "*/statsmodels/setup.cfg" -delete || die
        find . -wholename "*/${P}/LICENSE.txt" -delete || die
        find . -wholename "*/${P}/setup.cfg" -delete || die
    popd > /dev/null || die
    use doc && local HTML_DOCS=( docs/build/html/. )
    if use examples; then
        docompress -x /usr/share/doc/${PF}/examples
        dodoc -r examples
    fi
    distutils-r1_python_install_all
}
```

still won't fix it, though it deletes all occurrences except:

```
quiethost /var/tmp/portage/dev-python/statsmodels-0.10.1/work # find . -wholename "*LICENSE.txt*"
./statsmodels-0.10.1-python2_7/lib/statsmodels/stats/libqsturng/LICENSE.txt
./statsmodels-0.10.1/statsmodels/multivariate/factor_rotation/LICENSE.txt
./statsmodels-0.10.1/statsmodels/stats/libqsturng/LICENSE.txt
./statsmodels-0.10.1/docs/sphinxext/LICENSE.txt
./statsmodels-0.10.1-python3_6/lib/statsmodels/stats/libqsturng/LICENSE.txt
quiethost /var/tmp/portage/dev-python/statsmodels-0.10.1/work # equery f statsmodels | ag LICENSE.txt
/usr/LICENSE.txt
/usr/lib64/python2.7/site-packages/statsmodels/LICENSE.txt
/usr/lib64/python2.7/site-packages/statsmodels/stats/libqsturng/LICENSE.txt
/usr/lib64/python3.6/site-packages/statsmodels/LICENSE.txt
/usr/lib64/python3.6/site-packages/statsmodels/stats/libqsturng/LICENSE.txt
```

**Comment by juippis** [comment]
You can use `cp` or `mv` in src_install, operating in `${D}` to fix these in worst case.

**Comment by TheChymera** [comment]
@juippis could you comment on the new version linked above?

**Comment by juippis** [comment]
I have already, but this was popped in my mail before the new one, so I arrived here first :)

**Pull request closed without merging**